accepted/0003-interactive-audit-resolver.md

```diff
@@ -0,0 +1,165 @@
+- `.npmrc` is not a good option because it doesn't live in the repository and being a dotfile is easy to miss, so a developer could overlook it affecting the audit. Also using it for the purpose of storing resolutions to project specific issues seems counter-intuitive.
+
+A separate file referred to as `audit-resolve.json` has the benefit of being single purpose, easy to track and audit, easy to version and migrate between versions of `npm` and comfortable for the users to review in git history and pull requests.
+
+**postpone, remove, investigate**
+Other options are helpers for more elaborate actions a developer would take when confronted with an issue they can't or don't want to fix. 
+
+Why is postpone useful at all? It's designed to build a secure development culture where one didn't yet form. Without it, a developer under time pressure would mark an issue as ignored with intention to un-mark it later. 
+While shipping with a known vulnerability is a bad practice, NPM's mission with the community should be to empower people to build more secure products and trust their skill and understanding of their project's particular needs. We should also aspire to help teams introduce more secure workflows effortlessly, so letting a build pass without risking compromising security long-term is a win. 
+
+Remove is only useful as a convenience. Imagine a developer introducing `npm audit` and having to go through tens of issues. If they notice one of the first issues is caused by a dependency they no longer use, instead of remembering to clean it up later, they can choose this option.
+
+Investigate option is there to aid the user and direct them towards fixing the issue upstream.
+This RFC doesn't cover potential future usecases where NPM could do things ranging from linking to resources on mitigation to generating local fixes or providing a marketplace of companies offering services fixing the issue for customers (business potential for NPM).
+
+### Alternatives
+
+Currently the only alternative is to manage the issues manually and keep track of the resolutions using conventions created by a team individually. Ignoring dev dependencies and certain level of severity will be provided for `npm audit` so it lowers the barrier of entry to using it in a CI system. 
+
+Paid alternatives for managing security of a node.js project are available, including Snyk, with focus on providing patches to their customers.
+
+## Implementation
+
+reference implementation https://github.com/npm/cli/pull/10
+
+prototype of a working tool (in production use) https://www.npmjs.com/package/npm-audit-resolver
+
+The implementation is, and should remain, runnable standalone as a separate package with minor wrapping code - useful for testing new features without bundling unfinished work with npm cli versions and therefore node.js
+
+### audit-resolve.json
```

Comment:
It would - to me - be extremely preferable to have a less scoped file. Whether this ends up being `package.json` or `audit.json`, I'd very strongly prefer to avoid adding one-off files. Personally, I'd prefer to be able to just have this in my `package.json` since that's a file I already actively maintain in an `audit` property but I can see the potential value in splitting this out into an `audit.json` that could also define additional audit things for the future.

Reply:
There's a section with the analysis of why the file is separate from package.json etc.  
As for `audit-resolve` vs `audit` I don't really mind and I also don't see much of a difference, but I get the sentiment. 

The tooling I built supports extending the file over time and there's already a not-yet-defined section 'rules' next to 'decisions' I introduced to the spec in the package and plan to use to automate some decision making. 

I strongly believe there's nothing blocking this file from getting used in the extent it's described here in the initial implementation and then growing in scope as we figure out more.